### DISCUSSION TITLE
Connection retry when failed

### DISCUSSION THREAD

**Discussion by JuBanaSh** [FreeOpcUa/opcua-asyncio#412] [Q&A]
Dear all,
I'm calling the client.py in a script to connect to a siemens PLC and get data from it.
The PC is starting faster than the plc and so, my python script starts before the OPC UA connection is ready.
I would like to retry to connect with a loop but it fails as the client.py raise exceptions.
Is there anyway to do it no matter when the connection as not started or it has been lost?
I'm new to python, so your help will be very valuable and appreciated.

import os, datetime, time
from goto import goto, label
from opcua import Client

#Boucle infinie
while True:

        label .début
        try:
                #Configuration de l'URL d'accès au server
                #url = 'opc.tcp://192.168.20.10:4840'
                url = 'opc.tcp://172.25.212.10:4840'
                #Etablissement de la connexion
                client = Client(url)
                client.connect()

        except ConnectionError:
                print('en attente de connection')
                client.disconnect()
                time.sleep(5)
                goto .début
                
        else:
                print('Client connecté')

**Comment by AndreasHeine** [comment]
tbh your in the wrong repo ^^ this is the asyncio version of python-opcua
https://github.com/FreeOpcUa/python-opcua/discussions

```python
import time
from opcua import Client

if __name__ == "__main__":
    attempt = 0
    client = Client("opc.tcp://localhost:4840/freeopcua/server/")
    while 1:
        try:
            client.connect()
            break
        except Exception as e:
            attempt += 1
            print(f"Connection attempt: {attempt}, Error: {e}")
            time.sleep(2)
```

**Reply by JuBanaSh** [reply]
Arf... Sorry again, I think I got lost.
Anyway, thanks a lot for the feedback and the code sample, I will try it.
